### PR TITLE
[iAd] Xcode 10 beta 3 support.

### PR DESCRIPTION
**Pull request by mandel-macaque**
I'd appreciate feedback on what to do with the pinvoke I added in the ignore file. Not sure about its correct place or what to do with it.

### CODE REVIEW

File: src/iad.cs [hunk 1]
```diff
@@ -295,4 +295,21 @@ interface iAdPreroll_AVPlayerViewController {
 		[Export ("cancelPreroll")]
 		void CancelPreroll ();
 	}
+
+	[iOS (12,0)]
+	[NoWatch]
+	[DisableDefaultCtor]
+	[BaseType (typeof(UIViewController))]
+	interface ADInterstitialAdPresentationViewController
+	{
```
Comment (dalexsoto) [inline src/iad.cs]: Style, one line up

File: src/iad.cs [hunk 2]
```diff
@@ -295,4 +295,21 @@ interface iAdPreroll_AVPlayerViewController {
 		[Export ("cancelPreroll")]
 		void CancelPreroll ();
 	}
+
+	[iOS (12,0)]
+	[NoWatch]
+	[DisableDefaultCtor]
+	[BaseType (typeof(UIViewController))]
```
Comment (dalexsoto) [inline src/iad.cs]: Space `typeof(UIViewController)` -> `typeof (UIViewController)`

File: src/iad.cs [hunk 3]
```diff
@@ -295,4 +295,20 @@ interface iAdPreroll_AVPlayerViewController {
 		[Export ("cancelPreroll")]
 		void CancelPreroll ();
 	}
+
+	[iOS (12,0)]
+	[NoWatch]
+	[DisableDefaultCtor]
+	[BaseType ( typeof(UIViewController))]
```
Comment (dalexsoto) [inline src/iad.cs]: minor: no space **before** `typeof` but **after** it

File: src/iAd/ADBannerView.cs [hunk 4]
```diff
@@ -0,0 +1,15 @@
+using System.Runtime.InteropServices;
+using CoreGraphics;
+using ObjCRuntime;
+
+namespace iAd {
+	public partial class ADBannerView {
+
+			[DllImport (Constants.iAdLibrary)]
+			static extern CGSize ADClampedBannerSize (CGSize size);
```
Comment (spouliot) [inline src/iAd/ADBannerView.cs]: no availability attribute ?
it's an existing type and a new API

File: tests/monotouch-test/iAd/BannerViewTest.cs [hunk 5]
```diff
@@ -43,6 +43,14 @@ public void InitWithFrame ()
 				// Width and Height are set by the Ad (e.g. 320 x 50 for the iPhone)
 			}
 		}
+		
+		[Test]
+		public void GetClampedBannerSize ()
+		{
+			SizeF size = new SizeF (2.0, 2.0);
+			var result = ADBannerView.GetClampedBannerSize (size);
```
Comment (spouliot) [inline tests/monotouch-test/iAd/BannerViewTest.cs]: likely to fail in old iOS versions

**Pull request merged**